src/main/java/frc/robot/commands/test/TestHeadingPID.java

```diff
@@ -0,0 +1,40 @@
+package frc.robot.commands.test;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.constants.DriveConstants;
+import frc.robot.subsystems.Drivetrain;
+
+public class TestHeadingPID extends CommandBase {
+  
+  Drivetrain m_drive;
+  
+  // TODO: return if in error bound
+  public TestHeadingPID(Drivetrain drive){
+    m_drive = drive;
+    addRequirements(m_drive);
+  }
+  
+  @Override
+  public void execute() {
+    m_drive.setAllOptimize(false);
+    m_drive.m_headingPIDOutput = m_drive.getRotationController().calculate(m_drive.getAngleHeading(), m_drive.getRequestedHeadingEntry().getDouble(0));
```

Comment:
we might use the heading PID for other stuff, and this should be a function in the drivetrain, rather than being public